pyuvdata/fhd_cal.py

```diff
@@ -0,0 +1,195 @@
+                     run_check_acceptability=True):
+        """
+        Read data from an FHD cal.sav file.
+
+        Args:
+            cal_file: The cal.sav file to read from.
+            obs_file: The obs.sav file to read from.
+            settings_file: The settings_file to read from. Optional, but very
+                useful for provenance.
+            raw: Option to use the raw (per antenna, per frequency) solution or
+                to use the fitted (polynomial over phase/amplitude) solution.
+                Default is True (meaning use the raw solutions).
+            extra_history: Optional string or list of strings to add to the
+                object's history parameter. Default is None.
+            run_check: Option to check for the existence and proper shapes of
+                parameters after reading in the file. Default is True.
+            check_extra: Option to check optional parameters as well as required
+                ones. Default is True.
+            run_check_acceptability: Option to check acceptable range of the values of
+                parameters after reading in the file. Default is True.
+        """
+
+        this_dict = readsav(cal_file, python_dict=True)
+        cal_data = this_dict['cal']
+
+        this_dict = readsav(obs_file, python_dict=True)
+        obs_data = this_dict['obs']
+
+        self.Nspws = 1
+        self.spw_array = np.array([0])
+
+        self.Nfreqs = int(cal_data['n_freq'][0])
+        self.freq_array = np.zeros((self.Nspws, len(cal_data['freq'][0])), dtype=np.float_)
+        self.freq_array[0, :] = cal_data['freq'][0]
+        self.channel_width = float(np.mean(np.diff(self.freq_array)))
+
+        # FHD only calculates one calibration over all the times.
+        # cal_data.n_times gives the number of times that goes into that one
+        # calibration, UVCal.Ntimes gives the number of separate calibrations
+        # along the time axis.
+        self.Ntimes = 1
+        time_array = obs_data['baseline_info'][0]['jdate'][0]
+        self.integration_time = np.round(np.mean(np.diff(time_array)) * 24 * 3600, 2)
+        self.time_array = np.array([np.mean(time_array)])
+
+        self.Njones = int(cal_data['n_pol'][0])
+        # FHD only has the diagonal elements (jxx, jyy) and if there's only one
+        # present it must be jxx
+        if self.Njones == 1:
+            self.jones_array = np.array([-5])
+        else:
+            self.jones_array = np.array([-5, -6])
+
+        self.telescope_name = obs_data['instrument'][0]
+
+        self.Nants_data = int(cal_data['n_tile'][0])
+        self.Nants_telescope = int(cal_data['n_tile'][0])
+        self.antenna_names = cal_data['tile_names'][0].tolist()
+        self.antenna_numbers = np.arange(self.Nants_telescope)
+        self.ant_array = np.arange(self.Nants_data)
+
+        self.set_sky()
+        self.sky_field = 'phase center (RA, Dec): ({ra}, {dec})'.format(
```

Comment:
I'm not familiar with FHD, so forgive the naive question. Are these dictionary keys "standard" keywords that appear in all files? Put another way, is it possible for us to get a KeyError here? Should we bother putting this in a try/except block?

Reply:
My understanding is that the fields will always be present, but that some of the fields can be Nones or empty strings. I'm iterating with Ian to figure that out and I'll put in the appropriate checks. Thanks!

Reply:
Ok, I checked with Ian and it sounds like we're safe here.